client/extensions/woocommerce/app/order/index.js

```diff
@@ -42,43 +45,90 @@ class Order extends Component {
 		if ( newProps.orderId !== this.props.orderId || newProps.siteId !== this.props.siteId ) {
 			this.props.fetchOrder( newProps.siteId, newProps.orderId );
 			this.props.fetchNotes( newProps.siteId, newProps.orderId );
-		} else if ( newProps.order && this.props.order && newProps.order.status !== this.props.order.status ) {
+		} else if (
+			newProps.order &&
+			this.props.order &&
+			newProps.order.status !== this.props.order.status
+		) {
 			// A status change should force a notes refresh
 			this.props.fetchNotes( newProps.siteId, newProps.orderId, true );
 		}
 	}
 
-	onUpdate = ( order ) => {
-		// Merge the new order updates into the existing order updates
-		this.setState( ( prevState ) => {
-			const updatedOrder = { ...prevState.order, ...order };
-			return { order: updatedOrder };
-		} );
-	}
+	// Put this order into the editing state
+	toggleEditing = () => {
+		const { siteId, orderId } = this.props;
+		if ( siteId ) {
+			this.props.editOrder( siteId, { id: orderId } );
+		}
+	};
 
+	// Clear this order's edits, takes it out of edit state
+	cancelEditing = () => {
+		const { siteId } = this.props;
+		this.props.clearOrderEdits( siteId );
+	};
+
+	// Saves changes to the remote site via API
 	saveOrder = () => {
-		this.props.updateOrder( this.props.siteId, this.state.order );
-	}
+		const { siteId, order } = this.props;
+		this.props.updateOrder( siteId, order );
+		this.props.clearOrderEdits( siteId );
+	};
 
 	render() {
-		const { className, isSaving, order, orderId, site, translate } = this.props;
-		if ( ! order ) {
+		const {
+			className,
+			hasOrderEdits,
+			isEditing,
+			isSaving,
+			order,
+			orderId,
+			site,
+			translate,
+		} = this.props;
+		if ( isEmpty( order ) ) {
 			return null;
 		}
 
 		const breadcrumbs = [
-			( <a href={ getLink( '/store/orders/:site/', site ) }>{ translate( 'Orders' ) }</a> ),
-			( <span>{ translate( 'Order %(orderId)s Details', { args: { orderId: `#${ orderId }` } } ) }</span> ),
+			<a href={ getLink( '/store/orders/:site/', site ) }>{ translate( 'Orders' ) }</a>,
+			<span>
+				{ translate( 'Order %(orderId)s Details', { args: { orderId: `#${ orderId }` } } ) }
+			</span>,
 		];
+
+		let button = [
```

Comment:
Heh, I was going to say maybe this should be _buttons_ but see that it can be singular button below. 6 one way, half dozen the other.